packages/frontend/src/components/EditorRightDrawer/Step.tsx

```diff
@@ -0,0 +1,276 @@
+import { EditorContext } from '@/contexts/Editor'
+import { StepDisplayOverridesContext } from '@/contexts/StepDisplayOverrides'
+import { StepExecutionsProvider } from '@/contexts/StepExecutions'
+import { StepExecutionsToIncludeContext } from '@/contexts/StepExecutionsToInclude'
+import { replacePlaceholdersForHelpMessage } from '@/helpers/flow-templates'
+import { useStepMetadata } from '@/hooks/useStepMetadata'
+
+import FlowStepConfigurationModal from '../FlowStepConfigurationModal'
+import { infoboxMdComponents } from '../MarkdownRenderer/CustomMarkdownComponents'
+
+type FlowStepProps = {
+  collapsed?: boolean
+  step: IStep
+  isLastStep: boolean
+  index?: number | null
+  onChange: (step: IStep) => void
+  onClose: () => void
+  onContinue?: () => void
+  onOpen: () => void
+  templateConfig?: IFlowTemplateConfig
+}
+
+// FIXME (ogp-weeloong): remove this; not needed since we already do validation in FlowSubstep.
+function generateValidationSchema(substeps: ISubstep[]) {
+  const fieldValidations = substeps?.reduce(
+    (allValidations, { arguments: args }) => {
+      if (!args || !Array.isArray(args)) {
+        return allValidations
+      }
+
+      const substepArgumentValidations: Record<string, BaseSchema> = {}
+
+      for (const arg of args) {
+        const { key, required, hiddenIf } = arg
+
+        // base validation for the field if not exists
+        if (!substepArgumentValidations[key]) {
+          substepArgumentValidations[key] = yup.mixed()
+        }
+
+        if (typeof substepArgumentValidations[key] === 'object') {
+          // if the field is required and not conditionally hidden, add the
+          // required validation
+          if (required && !hiddenIf) {
+            substepArgumentValidations[key] = substepArgumentValidations[
+              key
+            ].required(`${key} is required.`)
+          }
+        }
+      }
+
+      return {
+        ...allValidations,
+        ...substepArgumentValidations,
+      }
+    },
+    {} as ObjectShape,
+  )
+
+  const validationSchema = yup.object({
+    parameters: yup.object(fieldValidations),
+  })
+
+  return yupResolver(validationSchema)
+}
+
+export default function Step(props: FlowStepProps): React.ReactElement | null {
+  const {
+    index,
+    step,
+    isLastStep,
+    collapsed,
+    onChange,
+    onContinue,
+    templateConfig,
+  } = props
+
+  const {
+    isOpen: isModalOpen,
+    onOpen: onModalOpen,
+    onClose: onModalClose,
+  } = useDisclosure()
+
+  const { onDrawerClose, testExecutionSteps } = useContext(EditorContext)
+  const displayOverrides = useContext(StepDisplayOverridesContext)?.[step.id]
+
+  const cannotChooseApp = displayOverrides?.disableActionChanges ?? false
+  const [currentSubstep, setCurrentSubstep] = useState<number | null>(
+    // OK to set to 1, even if a step has _no_ substeps, everything will just be
+    // collapsed due to matching logic below.
+    cannotChooseApp ? 1 : 0,
+  )
```

Comment:
```suggestion
  const [currentSubstep, setCurrentSubstep] = useState<number | null>(0)
```
Because i think not needed for non editor flowstep